kong/db/strategies/cassandra/connector.lua

```diff
@@ -14,7 +14,77 @@ function CassandraConnector.new(kong_config)
   do
     -- Resolve contact points before instantiating cluster, since the
     -- driver does not support hostnames in the contact points list.
+
+    -- TODO: this is an ugly hack to force lua sockets on a third party library
+
+    local tcp_old = ngx.socket.tcp
+    local udp_old = ngx.socket.udp
+
+    local dns_no_sync_old = kong_config.dns_no_sync
+
+    package.loaded["socket"] = nil
+    package.loaded["kong.tools.dns"] = nil
+    package.loaded["resty.dns.client"] = nil
+    package.loaded["resty.dns.resolver"] = nil
+
+    ngx.socket.tcp = function(...)
+      local tcp = require("socket").tcp(...)
+      return setmetatable({}, {
+        __newindex = function(_, k, v)
+          tcp[k] = v
+        end,
+        __index = function(_, k)
+          if type(tcp[k]) == "function" then
+            return function(_, ...)
+              if k == "send" then
+                local value = select(1, ...)
+                if type(value) == "table" then
+                  return tcp.send(tcp, table.concat(value))
+                end
+
+                return tcp.send(tcp, ...)
+              end
+
+              return tcp[k](tcp, ...)
+            end
+          end
+
+          return tcp[k]
+        end
+      })
+    end
+
+    ngx.socket.udp = function(...)
```

Comment:
Note for posterity:

ideally we should add support for `ngx.socket.udp` to lua-resty-socket and avoid duplicating the fallback logic there and here (lua-resty-socket has its own test suite and benefits other projects).

Reply:
Oh yes, looks like it already has all thing covered. Lets add dependency to it with 1.1+.